libs/networking/Testing_client.ml

```diff
@@ -0,0 +1,148 @@
+(* Austin Theriault
+ *
+ * Copyright (C) 2019-2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* Commentary *)
+(* This is a testing client to mock out http_helpers.ml so we
```

Comment:
better to put that in the .mli.
In theory people should just read the interface file (.mli) to understand how to use your library.
They can look at the .ml to understand the implementation, get possibly comments about design decisions.